spec/controllers/invoices/pdfs_controller_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+describe Invoices::PdfsController do
+  let(:customer) { create :customer, name: 'Hans' }
+  let(:employee) { create :employee }
+  let!(:invoice) { create :invoice, customer: customer, employee: employee, date: '2017-09-04' }
+
+  before do
+    sign_in employee
+  end
+
+  describe 'GET #show' do
+    let(:pdf_double) { instance_double Invoices::PDF }
+    let(:pdf_data) { double('data') }
+
+    before do
+      allow(Invoices::PDF).to receive(:new).and_return pdf_double
+      allow(pdf_double).to receive(:read).and_return pdf_data
+      allow(pdf_double).to receive(:persisted?).and_return true
+      allow(pdf_double).to receive(:persisted_pdf_path).and_return 'persisted.pdf'
+    end
+
+    it 'returns pdf path' do
+      expect(controller).to receive(:send_data).with(pdf_data, filename: 'Rechnung Hans 2017-09-04.pdf')
+
+      get :show, format: :pdf, params: { invoice_id: invoice.to_param }
```

Comment:
RSpec/MessageSpies: Prefer `have_received` for setting message expectations. Setup `controller` as a spy using `allow` or `instance_spy`.